docs/testing-deployment-publishing/running-your-app/local-run.md

```diff
@@ -0,0 +1,381 @@
+
+- To install Visual Code with Flutter and Dart plugins, check out [this link](https://flutter.dev/docs/get-started/editor?tab=vscode).
+- To install Android Studio with Flutter and Dart plugins, check out [this link](https://flutter.dev/docs/get-started/editor?tab=androidstudio).
+
+### 4. Running app on device
+
+You can choose to run your app on a real device or an emulator.
+
+To run your app on a device:
+
+1. First open the downloaded project in your preferred IDE.
+2. For **VS Code**:
+    1. Go to the "View" menu -> select "Terminal" from the dropdown.
+    2. Run the command `flutter pub get`.
+    3. Now, enter the command `flutter run`. VS Code will build and run your app. You'll see the output in the terminal, and the app should launch in the selected emulator or physical device.
+3. For **Android Studio**:
+    1. Open the terminal within Android Studio by clicking "View" -> "Tool Windows" -> "Terminal."
+    2. Run the command `flutter pub get`.
+    3. Click the green "Run" button (a right-facing triangle) located in the top toolbar. Choose the target device (emulator or physical device) where you want to run the app. Android Studio will build and run your app. You'll see the output in the "Run" panel at the bottom, and the app should launch in the selected emulator or device.
+
+:::info
+- If your device is not listed in the **Flutter Device Selection** dropdown, make sure you have properly completed the Android and iOS setup.
+- If you encounter a version compatibility issue with Flutter, you can resolve it by upgrading to the latest version. Simply execute the `flutter upgrade` command in your terminal. To verify your current Flutter version, use the `flutter --version` command.
+:::
+
+## Run on Desktop
+
+Running your app on a Desktop involves:
+
+1. **Adding platforms**: Navigate to **Setting and Integrations** from the Navigation Menu > **Project Setup** > **Platforms** and enable your desired platform.
+2. **Make design adjustments (optional)**: If you plan to target both mobile and desktop users, some design adjustments may be necessary to ensure that the UI is optimized for both platforms. You can create separate widgets for different platforms and control their visibility using [Responsive Visibility](#).
+3. **Run the app on a desktop**: Use the Local Run feature in the FlutterFlow Desktop app or manually download and run the code, choosing your target device (e.g., macOS) before running.
+
+## Video guide
+
+If you prefer watching a video tutorial, here's the one for you:
+<div class="video-container"><iframe width="630" height="355" src="https://www.youtube.com/embed/k9NpYncXC_U" title="Local Run | New Feature Tutorial" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe></div>
+
+---
+
+## Troubleshooting
```

Comment:
Can we move the troubleshooting questions under expanded sections like FAQs so its easier to navigate and looks cleaner?

Reply:
Done!